pkg/deploy/deploy.go

```diff
@@ -233,8 +233,15 @@ func manageResource(ctx context.Context, cli client.Client, obj *unstructured.Un
 			}
 		}
 
-		if obj.GetOwnerReferences() == nil {
+		existingOwnerReferences := obj.GetOwnerReferences()
+		if existingOwnerReferences == nil {
 			return cli.Delete(ctx, found)
+		} else if len(existingOwnerReferences) > 0 {
+			for _, owner := range existingOwnerReferences {
+				if owner.Kind != "DataScienceCluster" && owner.Kind != "DataScienceInitialization" {
+					return nil
+				}
+			}
```

Comment:
I'm trying to grasp the logic here.

- if owned by others - leave
- if there is no owner - delete
- if we own it
   - remove the old owner references
   - patch
   - delete

For the last scenario, could you help me understand the reason for patching before deleting? I'd love to get a clearer picture. Thanks!

Reply:
If deleted without cleaning up the owner references, get an error that resource is managed by `DSC` instance

Reply:
Ah right! Totally missed that. Maybe worth putting a short comment why we patch first so I won't come back with such questions anymore :)

Reply:
Yep, updated